### PR TITLE
[RSDK-3564] Make a modular component for a Periph board

### PR DESCRIPTION
**Pull request by penguinland**
Tried it on my raspberry pi: things work great as long as you specify the pins as the GPIO number, rather than the physical pin number (e.g., [the last pin](https://pinout.xyz/) is pin 21, not pin 40). 

This PR is much shorter than it looks: 1600 lines are `go.sum` and friends, and 800 are copied from the components/board/genericlinux/ directory within the RDK repo. It might be useful to review commit-by-commit, though the first few commits are just a "hello world" type thing with a fake sensor, instead of a board.

I also intend to try this out on an Orange Pi Zero 2 before merging...

### CODE REVIEW

**Comment by biotinker** [comment]
Weird that that didn't work for you.

I should have some time today or early next week to test this.

**Comment by penguinland** [comment]
I got it to work! The trick is to use the GPIO number, rather than the physical pin number. For example, [on the raspberry pi](https://pinout.xyz/), the third-to-last pin on the right of the diagram is pin 16, not pin 36. and with that, things work great on my pi! (or at least, pins 16 and 20 work great. Don't touch pin 38; it'll lock everything up. :upside_down_face: 

Thanks to Peter for this advice; it wouldn't have occurred to me on my own.

**Comment by penguinland** [comment]
Hm... I'm trying this on an Orange Pi Zero 2, and it runs, but I'm unable to control the pins I was expecting. It's possible I'm using the wrong names for the pins? I tried 9, hoping to get the one labeled PC9 in the user guide (third over from the 3.3 volt rail, not used in I2C), but pin 9 appears to be input-only, and always reads low even if I plug that physical pin to a high signal. Maybe it's looking at pin PH9 (second from the opposite corner of that same header) instead? but app.viam.com doesn't let me type non-numbers into the GPIO pin names (sounds like a bug, but a separate topic. If relevant, I'll file a bug later). I also tried pin 7 (hoping to get that same one, labeled PC9 but it's the 7th physical pin on the header), but with the same results.

Given that I can get the RDK to start up at all and give back errors that the pins aren't doing what I expected, it seems like it's able to create the component and send/receive commands, which makes me think this PR is working as intended.

**Comment by biotinker** [comment]
"9" is not a valid pin, nor do you want to use the "PH#" value.

Here's is the table for the orangepi zero 2; you want the number in the "GPIO No" column https://f1atb.fr/wp-content/uploads/2021/05/pinzero2g.jpg

**Comment by penguinland** [comment]
Thanks, pin 73 outputs like a champ! The "Get pin state" button gives back what the output has been set to most recently, rather than reading the pin as an input. This matches with something Steve has said before, but not with our raspberry pi implementation or our non-Periph GPIO implementation. I'll investigate what's wrong there...

**Comment by penguinland** [comment]
Okay, this time it works as intended! See the 2 most recent commits. but this also suggests that historically we haven't been able to use GPIO pins as inputs, which makes me wonder if this is a breaking change for anyone who uses a Periph board. @biotinker, LMK if this is a problem for you.

File: periphboard/periph_gpio.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ func (gp periphGpioPin) set(high bool) error {
 }
 
 func (gp periphGpioPin) Get(ctx context.Context, extra map[string]interface{}) (bool, error) {
+	gp.pin.In(gpio.PullNoChange, gpio.NoEdge)
```
Comment (penguinland) [inline periphboard/periph_gpio.go]: sigh. This can return an error. Check it

**Comment by penguinland** [comment]
@mcvella had asked me to look into whether the I2C and SPI buses still work. On further consideration, though, they should be removed entirely: since this is a modular component, no other component outside of this module can use the classes/structs in here to access I2C/SPI devices. The approximate plan as we make more modular components is that each component should be able to access those buses on their own, without going through a board component. So, I should actually remove that code!

FWIW, to get the I2C bus set up on the Orange Pi Zero2, I needed to edit `/boot/orangepiEnv.txt` to add the line `overlays=i2c3` (and then reboot)

**Review by biotinker — APPROVED** [review]
Haven't had a chance to test but lgtm

**Pull request merged**